code/modules/spells/spell_types/touch/invisible_touch.dm

```diff
@@ -0,0 +1,95 @@
+#define MIME_INVISIBLE_TOUCH_LENGTH 16 SECONDS
+
+/datum/action/cooldown/spell/touch/invisible_touch
+	name = "Invisible Touch"
+	desc = "Touch something to make it disappear temporarily."
+	background_icon_state = "bg_mime"
+	overlay_icon_state = "bg_mime_border"
+	button_icon = 'icons/obj/wizard.dmi'
+	button_icon_state = "nothing"
+	panel = "Mime"
+
+	invocation = ""
+	invocation_type = INVOCATION_EMOTE
+	check_flags = AB_CHECK_CONSCIOUS|AB_CHECK_HANDS_BLOCKED|AB_CHECK_INCAPACITATED
+
+	spell_requirements = SPELL_REQUIRES_HUMAN|SPELL_REQUIRES_MIME_VOW
+	antimagic_flags = NONE
+
+	school = SCHOOL_MIME
+	cooldown_time = MIME_INVISIBLE_TOUCH_LENGTH * 2
+	spell_max_level = 1
+
+	hand_path = /obj/item/melee/touch_attack/invisible_touch
+	draw_message = span_notice("You channel mime power into your hand.")
+	drop_message = span_notice("You let the power from your hand dissipate.")
+
+	var/list/things = list()
+	var/list/blacklist = list (
+        /obj/machinery/nuclearbomb/selfdestruct,
+        /obj/structure/closet
+        )
+
+/datum/action/cooldown/spell/touch/invisible_touch/is_valid_target(atom/cast_on)
+	return TRUE
```

Comment:
```suggestion
	return (!isturf(cast_on)
```
this way it won't get wasted on turf

Reply:
Interesting thing: This was tried. I do not know why but it didn't work so I moved the check elsewhere

Reply:
See line 37-38

Reply:
Its because touch is currently bugged, i have a pr fixing it

Reply:
You are fixing a completely different proc that doesn't even call this

Reply:
i'm fixing a proc that calls a proc that calls this

Reply:
Well tell me after it is merged